README.md

```diff
@@ -1,26 +1,230 @@
+
+```ocaml
+  type 'a cont
+
+  val suspend : ('a cont -> 'a option) -> 'a
+  val resume : 'a cont -> 'a -> unit
+  val get_tid : unit -> int
 ```
-$ opam switch create 5.0.0
-$ git clone git@github.com:ocaml-multicore/reagents.git
+
+If a running reagent cannot make progress, the fiber is automatically suspended.
+Once someone else updates relevant state, Reagents trigger required resumptions.
+A toy scheduler is available in [Reagents.Toy_scheduler](lib/toy_scheduler.mli).
+
+### Type
+
+A computation within reagents framework has the following type
+`('a, 'b) Reagents.t`. Here, the computation takes a parameter of type `'a` and
+returns value of type `'b`. Internally, it may consist of any number of
+operations and do any number of side effects. Crucially, regardless of the
+construction of a reagent, all of its operations execute atomically and entirely
+or none at all.
+
+### Combinators
+
+Reagents can be composed in arbitrary ways. The following three main combinators
+are exposed by the library.
+
+```ocaml
+(* Sequential composition runs one reagent after another. It passes the result of the first one as parameter to the second one. *)
+val (>>>) : ('a,'b) t -> ('b,'c) t -> ('a,'c) t
+(* Conjunction executes both reagents at once and returns both results. Note, this combinator still attempts to execute its components sequentially. It only differs with [>>>] in information flow. *)
+val (<*>) : ('a,'b) t -> ('a,'c) t -> ('a,'b * 'c) t
+(* Disjunction tries to execute the first reagent and if it blocks, it attempts the second one. If both block, the first one to unblock is executed. *)
+val (<+>) : ('a,'b) t -> ('a,'b) t -> ('a,'b) t
 ```
 
-## Documentation
+### Others
 
-The documentation and API reference is automatically generated from 
-the source interfaces. It can be consulted [online](https://ocaml-multicore.github.io/reagents/doc/).
+There is a number of other values defined in the
+[public interface](lib/base_intf.ml) that serve as units, helpers or
+transformations for existing reagents. Perhaps the most notable one is
+`attempt`, which converts a blocking reagent into non-blocking one.
+
+```ocaml
+val attempt : ('a, 'b) t -> ('a, 'b option) t
+(** Convert a blocking reagent into a non-blocking one. If reagent [r] is a
+    blocks, then [attempt r] return [None]. If [r] does not block and returns
+    a value [v], then [attempt r] returns [Some v]. *)
+```
+
+### Executing a reagent
+
+Once a desired reagent has been composed, it can be run.
+
+```ocaml
+val run : ('a, 'b) t -> 'a -> 'b
+(** [run r v] runs the reagents [r] with value [v]. *)
+
+```
+
+Note, this function has to be executed from within a scheduler for the
+suspension and resumption effects to be handled correctly.
+
+### Data structures
+
+Reagents exposes two core data structures. Complex data structures should
+utilise these as buildling blocks, if possible.
+
+- Reference - a low-level object akin to an `'a Atomic.t`, which can be modified
```

Comment:
I'd perhaps use an `&mdash;` instead of - or maybe emphasize the concept name.

- *Reference* &mdash; a low-level object ...
- *Channel* &mdash; a two-way channel ...

Basically adding visual redundancy.

Reply:
Indeed, looks much nicer!